### PR TITLE
implement handling for alias field resolving in more cases

### PR DESCRIPTION
**Pull request by sokra**
### Description

Implement a few more edge cases for resolving the alias field

### Testing Instructions

Closes PACK-2605

### CODE REVIEW

File: crates/turbopack-tests/tests/execution/turbopack/resolving/alias-field/input/node_modules/package/package.json [hunk 1]
```diff
@@ -0,0 +1,9 @@
+{
+  "browser": {
+    "./dir/file1.js": "./replaced/file1.js",
+    "./dir/file2": "./replaced/file2",
+    "dir/file3.js": "./replaced/file3.js",
```
Comment (kdy1) [inline crates/turbopack-tests/tests/execution/turbopack/resolving/alias-field/input/node_modules/package/package.json:5]: I'm not sure if this is correct. I'll take a look at the node.js doc

Comment (sokra) [reply crates/turbopack-tests/tests/execution/turbopack/resolving/alias-field/input/node_modules/package/package.json:5]: It's not the the spec, and I don't think it's "correct", but webpack supported that and that helps with compatibility

Comment (kdy1) [reply crates/turbopack-tests/tests/execution/turbopack/resolving/alias-field/input/node_modules/package/package.json:5]: I see. Thank you for the explanation!

**Review by kdy1 — APPROVED** [review]
